src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs

```diff
@@ -0,0 +1,121 @@
+                    .Create();
+
+                schemeNotMatchingComplianceYear = SchemeDbSetup.Init()
+                    .WithAuthority(authority1.Id)
+                    .WithOrganisation(organisation.Id)
+                    .WithStatus(SchemeStatus.Approved)
+                    .Create();
+
+                var obligationUploadCurrentYear = ObligationUploadDbSetup.Init().Create();
+                var obligationUploadNotMatchingYear = ObligationUploadDbSetup.Init().Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(scheme1.Id)
+                    .WithObligationUpload(obligationUploadCurrentYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(scheme2.Id)
+                    .WithObligationUpload(obligationUploadCurrentYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(schemeNotMatchingComplianceYear.Id)
+                    .WithObligationUpload(obligationUploadNotMatchingYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year - 1)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(schemeNotMatchingComplianceYear.Id)
+                    .WithObligationUpload(obligationUploadNotMatchingYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year + 1)
+                    .Create();
+
+                request = new GetSchemesWithObligation(SystemTime.UtcNow.Year);
+            };
+
+            private readonly Because of = () =>
+            {
+                result = Task.Run(async () => await handler.HandleAsync(request)).Result;
+            };
+
+            private readonly It shouldHaveReturnedObligationYears = () =>
+            {
+                result.Should().NotBeNull();
+                result.Count.Should().Be(2);
+                result.Should().Contain(s => s.Id == scheme1.Id);
+                result.Should().Contain(s => s.Id == scheme2.Id);
+                result.Should().OnlyHaveUniqueItems();
+            };
+        }
+
+        public class GetSchemesWithObligationHandlerTestBase : WeeeContextSpecification
+        {
+            protected static IRequestHandler<GetSchemesWithObligation, List<SchemeData>> handler;
+            protected static GetSchemesWithObligation request;
+            protected static Fixture fixture;
+            protected static List<SchemeData> result;
+
+            public static IntegrationTestSetupBuilder LocalSetup()
+            {
+                var setup = SetupTest(IocApplication.RequestHandler)
+                    .WithIoC()
+                    .WithTestData(true)
+                    .WithInternalUserAccess();
+
+                var authority = Query.GetEaCompetentAuthority();
+                var role = Query.GetAdminRole();
```

Comment:
do you need that ? the handler has only    authorization.EnsureCanAccessInternalArea(), doesn't check for the admin role?

Reply:
Updated...